### PR TITLE
pkg: Support SHA field with Alertmanager and Thanos

### PR DESCRIPTION
**Pull request by metalmatze**
This is a follow up of #1847 and will add a SHA field to AlertmanagerSpec and ThanosSpec to specify a container image by its SHA sum.

### CODE REVIEW

File: example/prometheus-operator-crd/alertmanager.crd.yaml [hunk 1]
```diff
@@ -1795,6 +1795,12 @@ spec:
               description: ServiceAccountName is the name of the ServiceAccount to
                 use to run the Prometheus Pods.
               type: string
+            sha:
+              description: SHA of Thanos container image to be deployed. Defaults
+                to the value of `version`. Similar to a tag, but the sha explicitly
+                deploys an immutable container image. Version anmaked Tag are ignored
```
Comment (mxinden) [inline example/prometheus-operator-crd/alertmanager.crd.yaml]: `s/anmaked/and/`

File: example/prometheus-operator-crd/alertmanager.crd.yaml [hunk 2]
```diff
@@ -1795,6 +1795,12 @@ spec:
...
+                if SHA is set.
```
Comment (squat) [inline example/prometheus-operator-crd/alertmanager.crd.yaml]: nit: there is a mix of `sha` and `SHA` in this sentence

**Comment by metalmatze** [comment]
Typo and wording are fixed. This is now good to merge from my side. @mxinden @squat

**Comment by squat** [comment]
thanks, @metalmatze

**Pull request merged**